app/models/package_manager/pypi.rb

```diff
@@ -123,14 +125,22 @@ def self.known_versions(name)
         &.index_by { |v| v[:number] } || {}
     end
 
-    def self.dependencies(name, version, _mapped_project)
+    # Simply parses out the name of a PEP 508 Dependency specification: https://peps.python.org/pep-0508/
+    # Leaves the rest as-is with any leading semicolons or spaces stripped
+    def self.parse_pep_508_dep_spec(dep)
+      name, requirement = dep.split(PEP_508_NAME_REGEX, 2).last(2).map(&:strip)
```

Comment:
worth noting that `requirement` is not necessarily a version still. I think it's still an improvement over what was here before though. To really be certain, I think we'll have to implement that parser and use it to see if we can pull a version out. I think that's beyond the scope of this ticket, but worth filing a separate story for.

Reply:
but `requirements` is what we need: https://github.com/librariesio/libraries.io/blob/main/app/models/dependency.rb#L12

It wouldn't be possible to pull a version out of there i don't think.

Reply:
got it, that makes sense. I think what confused me is it being named `version` in `self.dependencies`. Worth renaming that to `requirement`/`requirements`